### PR TITLE
Features/inv 73 reusable

### PR DESCRIPTION
**Pull request by stegar123**
# Description
(REQUIRED)
Describe what the new code is used for.
CI/CD packaging rpm and deb for package packetfence-perl

### CODE REVIEW

File: .github/workflows/reusable_unit_test.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          path:  /mnt
+        env:
+          ARTIFACTORY_NAME: ${{ inputs._IMAGE_TYPE == 'rhel8' && 'package-rpm' || 'package-deb' }}
+
+      - name: Install the package packetfence-perl ${{ inputs._IMAGE_TYPE }}
+        run: |
+          ls -la /mnt
+          ${{ inputs._IMAGE_TYPE == 'rhel8' && env.INSATALL_RPM_PACKAGE  || env.INSALL_DEB_PACKAGE }}
+        env:
+          INSATALL_RPM_PACKAGE: "set -x && yum -y install findutils && find /mnt -name ${{ inputs._PACKAGE_NAME }}*.rpm -exec rpm -ivh {} \\;"
```
Comment (fdurand) [inline .github/workflows/reusable_unit_test.yml]: INSTALL_RPM_PACKAGE

File: .github/workflows/reusable_unit_test.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          INSALL_DEB_PACKAGE: "set -x && find /mnt -name ${{ inputs._PACKAGE_NAME }}*.deb -exec dpkg -i {} \\;"
```
Comment (fdurand) [inline .github/workflows/reusable_unit_test.yml]: INSTALL_DEB_PACKAGE

File: .github/workflows/reusable_unit_test.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        with:
+          name: ${{ env.ARTIFACTORY_NAME }}
+          path:  /mnt
+        env:
+          ARTIFACTORY_NAME: ${{ inputs._IMAGE_TYPE == 'rhel8' && 'package-rpm' || 'package-deb' }}
+
+      - name: Install the package packetfence-perl ${{ inputs._IMAGE_TYPE }}
+        run: |
+          ls -la /mnt
+          ${{ inputs._IMAGE_TYPE == 'rhel8' && env.INSATALL_RPM_PACKAGE  || env.INSALL_DEB_PACKAGE }}
```
Comment (fdurand) [inline .github/workflows/reusable_unit_test.yml]: Wrong wording

File: containers/packetfence-perl/debian/Dockerfile_debian [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    apt install libmodule-signature-perl zip make build-essential \ 
+libssl-dev zlib1g-dev libmariadb-dev-compat libmariadb-dev libssh2-1-dev libexpat1-dev \
+pkg-config libkrb5-dev libsystemd-dev libgd-dev libcpan-distnameinfo-perl libyaml-perl \
+curl wget graphviz libio-socket-ssl-perl debhelper \ 
+libnet-ssleay-perl libcpan-perl-releases-perl python3 vim -y && \
+    apt-get clean && \ 
+    rm -rf /var/lib/apt/lists/* && \
+    rm -rf /tmp/* /var/tmp/*
+
+#RUN TestPerlConfig=$(perl -e exit); [[ ${TestPerlConfig} != "" ]] && export LC_CTYPE=en_US.UTF-8;  export LC_ALL=en_US.UTF-8
```
Comment (fdurand) [inline containers/packetfence-perl/debian/Dockerfile_debian]: If not used then remove the line or add a comment before it to explain why it's here.

File: containers/packetfence-perl/rhel8/Dockerfile_rhel8 [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    perl-CPAN perl-IO-Socket-SSL perl-Net-SSLeay perl-Devel-Peek perl-CPAN-DistnameInfo && \ 
+    dnf group install -y "Development Tools"
+
+RUN subscription-manager repos --enable codeready-builder-for-rhel-8-x86_64-rpms && \
+    dnf -y module enable virt-devel && \
+    dnf install -y libssh2-devel
+
+RUN subscription-manager unregister
+
+#RUN TestPerlConfig=$(perl -e exit); [[ ${TestPerlConfig} != "" ]] && export LC_CTYPE=en_US.UTF-8;  export LC_ALL=en_US.UTF-8
```
Comment (fdurand) [inline containers/packetfence-perl/rhel8/Dockerfile_rhel8]: If not used then remove the line or add a comment before it to explain why it's here.

File: .github/workflows/main_packetfence-perl.yml [hunk 6]
```diff
@@ -0,0 +1,115 @@
+name: packetfence-perl_debian_rhel8_package
+on:
+  workflow_dispatch:
+  push:
+    branches-ignore: 
+      - 'devel'   # excludes devel
```
Comment (fdurand) [inline .github/workflows/main_packetfence-perl.yml]: Why ignoring devel ?

File: .github/workflows/packetfence-perl_build_image_package.yml [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+      - name: Find package version
+        id: contents
+        run: |
+          package_version=$(${{ inputs._IMAGE_TYPE == 'rhel8' &&  env.command_rpm || env.command_deb}})
+          echo "Found version: $package_version"
+          echo "PACKAGE_VERSION=$package_version" >> "$GITHUB_ENV"
+          echo "data=$package_version" >> "$GITHUB_OUTPUT"
+        env: 
+          command_rpm: set -e && docker run --rm -i -v ${{ github.workspace }}/addons/packetfence-perl:${{ inputs._WORKDIR }}/ rockylinux:8.8 /bin/bash -c \
```
Comment (fdurand) [inline .github/workflows/packetfence-perl_build_image_package.yml]: Do we have to be consistent , like use real rhel8 ?

File: .github/workflows/reusable_upload_packages.yml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        rsync -avzr --delete --rsync-path='mkdir -p  ${{ env.PACKAGE_DEST_PATH }} && rsync' -e "ssh -o StrictHostKeyChecking=no -p 22"  ${{ inputs._VOLUME_MOUNT }}/${{ inputs._IMAGE_TYPE }}/packages/${{inputs._PACKAGE_NAME}}${{ env.PACKAGE_TYPE }} root@192.95.20.194:${{ env.PACKAGE_DEST_PATH }}
+        ssh-add -D
+      env:
+        PACKAGE_TYPE: ${{ inputs._IMAGE_TYPE == 'rhel8' &&  '*.rpm' || '*.deb' }}
+
+
+    - name: Remove files
+      uses: addnab/docker-run-action@v3
+      with:
+        image: rockylinux:8.8
```
Comment (fdurand) [inline .github/workflows/reusable_upload_packages.yml]: Do we have to be consistent , like use real rhel8 ?

File: containers/packetfence-perl/debian/Dockerfile_debian [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+ENV PERL5LIB=/root/perl5/lib/perl5:${BASE_DIR}/lib/perl5/ \
+    PKG_CONFIG_PATH=/usr/lib/pkgconfig/ \
+    CPAN_BIN_PATH="/usr/bin/cpan" \
+    CPAN_VERSION=2.36
+
+WORKDIR ${WORKDIR}
+
+RUN  apt update -y && \
+    apt install libmodule-signature-perl zip make build-essential \
```
Comment (fdurand) [inline containers/packetfence-perl/debian/Dockerfile_debian:19]: can we do a apt upgrade here ?

File: containers/packetfence-perl/rhel8/Dockerfile_rhel8 [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
+RUN chmod +x /tini
+
+WORKDIR ${WORKDIR}
+
+COPY ./addons/packetfence-perl/dependencies.csv ./addons/packetfence-perl/build_package.sh ./addons/packetfence-perl/install_cpan.py ./addons/packetfence-perl/psono.py ${WORKDIR}
+
+#Install python module
+RUN dnf -y  install python39 && python3.9 -m pip install -q  -U pip && pip install -q pynacl requests
+
```
Comment (fdurand) [inline containers/packetfence-perl/rhel8/Dockerfile_rhel8:26]: Same as Debian, can we upgrade the os ?

File: .github/workflows/main_packetfence-perl.yml [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+      - name: Declare 'needs' variable
+        id: variable
+        run: |
+          echo "volume_mount=${{env.VOLUME_MOUNT}}" >> "$GITHUB_OUTPUT"
+          echo "workdir=${{env.WORKDIR}}" >> "$GITHUB_OUTPUT"
+          echo "output_directory=${{env.OUTPUT_DIRECTORY}}" >> "$GITHUB_OUTPUT"
+          echo "branch_name=${{env.BRANCH_NAME}}" >> "$GITHUB_OUTPUT"
+          echo "package_name=${{env.PACKAGE_NAME}}" >> "$GITHUB_OUTPUT"
+
+  buil_images_and_packages:
```
Comment (satkunas) [inline .github/workflows/main_packetfence-perl.yml]: build_images_and_packages

**Review by satkunas — CHANGES_REQUESTED** [review]

File: .github/workflows/main_packetfence-perl.yml [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+        id: variable
+        run: |
+          echo "volume_mount=${{env.VOLUME_MOUNT}}" >> "$GITHUB_OUTPUT"
+          echo "workdir=${{env.WORKDIR}}" >> "$GITHUB_OUTPUT"
+          echo "output_directory=${{env.OUTPUT_DIRECTORY}}" >> "$GITHUB_OUTPUT"
+          echo "branch_name=${{env.BRANCH_NAME}}" >> "$GITHUB_OUTPUT"
+          echo "package_name=${{env.PACKAGE_NAME}}" >> "$GITHUB_OUTPUT"
+
+  build_images_and_packages:
+    if: ${{ contains( github.event.head_commit.message, '[perl]') || needs.build_prepation.outputs.path_changes == 'true' && ( needs.build_prepation.outputs.branch_name == 'devel' || needs.build_prepation.outputs.branch_name == 'maintenance/13.0' ) }}
```
Comment (JeGoi) [inline .github/workflows/main_packetfence-perl.yml]:
```suggestion
    if: ${{ contains( github.event.head_commit.message, '[perl]') || needs.build_preparation.outputs.path_changes == 'true' && ( needs.build_preparation.outputs.branch_name == 'devel' || needs.build_preparation.outputs.branch_name == 'maintenance/13.0' ) }}
```

File: .github/workflows/main_packetfence-perl.yml [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+
+env:
+  WORKDIR: "/root"
+  VOLUME_MOUNT: "/mnt/${{ github.run_id }}_${{ github.run_attempt }}"
+  OUTPUT_DIRECTORY: "/mnt/output/"
+  BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+  PACKAGE_NAME: 'packetfence-perl'
+
+jobs:
+  build_prepation:
```
Comment (JeGoi) [inline .github/workflows/main_packetfence-perl.yml]:
```suggestion
  build_preparation:
```

File: .github/workflows/main_packetfence-perl.yml [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+          echo "branch_name=${{env.BRANCH_NAME}}" >> "$GITHUB_OUTPUT"
+          echo "package_name=${{env.PACKAGE_NAME}}" >> "$GITHUB_OUTPUT"
+
+  build_images_and_packages:
+    if: ${{ contains( github.event.head_commit.message, '[perl]') || needs.build_prepation.outputs.path_changes == 'true' && ( needs.build_prepation.outputs.branch_name == 'devel' || needs.build_prepation.outputs.branch_name == 'maintenance/13.0' ) }}
+    strategy:
+      matrix:
+        images: ['debian', 'rhel8']
+    uses:  ./.github/workflows/packetfence-perl_build_image_package.yml
+    needs: ['build_prepation']
```
Comment (JeGoi) [inline .github/workflows/main_packetfence-perl.yml]:
```suggestion
    needs: ['build_preparation']
```

File: .github/workflows/main_packetfence-perl.yml [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+    with:
+      _IMAGE_TYPE: ${{ matrix.images }}
+      _WORKDIR:  ${{needs.build_prepation.outputs.workdir}}
+      _OUTPUT_DIRECTORY: ${{needs.build_prepation.outputs.output_directory}}
+      _BRANCH_NAME: ${{ needs.build_prepation.outputs.branch_name }}
```
Comment (JeGoi) [inline .github/workflows/main_packetfence-perl.yml]:
```suggestion
      _WORKDIR:  ${{needs.build_preparation.outputs.workdir}}
      _OUTPUT_DIRECTORY: ${{needs.build_preparation.outputs.output_directory}}
      _BRANCH_NAME: ${{ needs.build_preparation.outputs.branch_name }}
```

File: .github/workflows/main_packetfence-perl.yml [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+    secrets: inherit
+  
+  unit_tests_packages:
+    strategy:
+      matrix:
+        images: ['debian', 'rhel8']
+    uses:  ./.github/workflows/reusable_unit_test.yml
+    needs: ['build_prepation', 'build_images_and_packages']
```
Comment (JeGoi) [inline .github/workflows/main_packetfence-perl.yml]:
```suggestion
    needs: ['build_preparation', 'build_images_and_packages']
```

File: .github/workflows/main_packetfence-perl.yml [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+    with:
+      _IMAGE_TYPE: ${{ matrix.images }}
+      _PACKAGE_NAME: ${{ needs.build_prepation.outputs.package_name }}
```
Comment (JeGoi) [inline .github/workflows/main_packetfence-perl.yml]:
```suggestion
      _PACKAGE_NAME: ${{ needs.build_preparation.outputs.package_name }}
```

File: .github/workflows/main_packetfence-perl.yml [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+
+  sign_package:
+    strategy:
+      matrix:
+        images: ['debian', 'rhel8']
+    uses:  ./.github/workflows/reusable_sign_packages.yml
+    needs: ['build_prepation', 'build_images_and_packages', 'unit_tests_packages']
```
Comment (JeGoi) [inline .github/workflows/main_packetfence-perl.yml]:
```suggestion
    needs: ['build_preparation', 'build_images_and_packages', 'unit_tests_packages']
```

File: .github/workflows/main_packetfence-perl.yml [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+    with:
+      _IMAGE_TYPE: ${{ matrix.images }}
+      _VOLUME_MOUNT: ${{needs.build_prepation.outputs.volume_mount}}
+      _BRANCH_NAME: ${{ needs.build_prepation.outputs.branch_name }}
+      _PACKAGE_NAME: ${{ needs.build_prepation.outputs.package_name }}
```
Comment (JeGoi) [inline .github/workflows/main_packetfence-perl.yml]:
```suggestion
      _VOLUME_MOUNT: ${{needs.build_preparation.outputs.volume_mount}}
      _BRANCH_NAME: ${{ needs.build_preparation.outputs.branch_name }}
      _PACKAGE_NAME: ${{ needs.build_preparation.outputs.package_name }}
```

File: .github/workflows/main_packetfence-perl.yml [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+    secrets: inherit
+
+  upload_packages:
+    strategy:
+      matrix:
+        images: ['debian', 'rhel8']
+    uses:  ./.github/workflows/reusable_upload_packages.yml
+    needs: ['build_prepation', 'build_images_and_packages', 'unit_tests_packages', 'sign_package']
```
Comment (JeGoi) [inline .github/workflows/main_packetfence-perl.yml]:
```suggestion
    needs: ['build_preparation', 'build_images_and_packages', 'unit_tests_packages', 'sign_package']
```

File: .github/workflows/main_packetfence-perl.yml [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+    with:
+      _IMAGE_TYPE: ${{ matrix.images }}
+      _VOLUME_MOUNT: ${{needs.build_prepation.outputs.volume_mount}}
+      _BRANCH_NAME: ${{ needs.build_prepation.outputs.branch_name }}
+      _PACKAGE_NAME: ${{ needs.build_prepation.outputs.package_name }}
```
Comment (JeGoi) [inline .github/workflows/main_packetfence-perl.yml]:
```suggestion
      _VOLUME_MOUNT: ${{needs.build_preparation.outputs.volume_mount}}
      _BRANCH_NAME: ${{ needs.build_preparation.outputs.branch_name }}
      _PACKAGE_NAME: ${{ needs.build_preparation.outputs.package_name }}
```

File: .github/workflows/reusable_sign_packages.yml [hunk 22]
```diff
@@ -0,0 +1,104 @@
+name: Signe debian package
```
Comment (JeGoi) [inline .github/workflows/reusable_sign_packages.yml]:
```suggestion
name: Sign Debian and REL packages
```

File: .github/workflows/reusable_sign_packages.yml [hunk 23]
```diff
@@ -0,0 +1,104 @@
...
+
+  sign_package_rpm:
+    runs-on: packetfence-perl-package-build 
+    if: inputs._IMAGE_TYPE == 'rhel8'
+    container:
+      image: rockylinux:8.8
+      volumes:
+        - ${{ github.workspace }}/addons/packetfence-perl/:/root
+    steps:
+      - name: Dwnload artifactory ${{ inputs._IMAGE_TYPE }}
```
Comment (JeGoi) [inline .github/workflows/reusable_sign_packages.yml]:
```suggestion
      - name: Download artifactory ${{ inputs._IMAGE_TYPE }}
```

File: .github/workflows/reusable_sign_packages.yml [hunk 24]
```diff
@@ -0,0 +1,104 @@
...
+        run:  apt -qq update && apt -qq -y install gpg dpkg-sig python3 python3-pip && python3 -m pip install -q  -U pip && pip install -q pynacl requests
+
+      - name: import private key ${{ inputs._IMAGE_TYPE }}
+        run: | 
+          set -e && python3 /root/psono.py --api_key_id=${{ secrets.PSONO_API_KEY_ID }} --api_key_secret_key=${{ secrets.PSONO_API_KEY_SECRET_KEY }} --secret_id=${{ vars.PSONO_GPG_PRIVATE_KEY_SECRET_ID }} --return_value=password_notes | gpg --pinentry-mode loopback --import
+
+      - name: Sign the package ${{ inputs._IMAGE_TYPE }}
+        run: dpkg-sig -k  B022C48D3D6373D7FC256A8CCB2D3A2AA0030E2C  --sign builder /mnt/${{ inputs._PACKAGE_NAME }}*.deb
+
+      - name: Verifiy the signarure of package ${{ inputs._IMAGE_TYPE }}
```
Comment (JeGoi) [inline .github/workflows/reusable_sign_packages.yml]:
```suggestion
      - name: Verify ${{ inputs._IMAGE_TYPE }} Package Signature
```

File: .github/workflows/reusable_sign_packages.yml [hunk 25]
```diff
@@ -0,0 +1,104 @@
...
+        with:
+          name: ${{ env.ARTIFACTORY_NAME }}
+          path:  /mnt
+        env:
+          ARTIFACTORY_NAME: ${{ inputs._IMAGE_TYPE == 'rhel8' && 'package-rpm' || 'package-deb' }}
+
+      - name: Install debian dependenecies ${{ inputs._IMAGE_TYPE }}
+        run:  apt -qq update && apt -qq -y install gpg dpkg-sig python3 python3-pip && python3 -m pip install -q  -U pip && pip install -q pynacl requests
+
+      - name: import private key ${{ inputs._IMAGE_TYPE }}
```
Comment (JeGoi) [inline .github/workflows/reusable_sign_packages.yml]:
```suggestion
      - name: Import private key ${{ inputs._IMAGE_TYPE }}
```

File: .github/workflows/reusable_sign_packages.yml [hunk 26]
```diff
@@ -0,0 +1,104 @@
...
+    steps:
+      - name: Download artifactory ${{ inputs._IMAGE_TYPE }}
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ env.ARTIFACTORY_NAME }}
+          path:  /mnt
+        env:
+          ARTIFACTORY_NAME: ${{ inputs._IMAGE_TYPE == 'rhel8' && 'package-rpm' || 'package-deb' }}
+
+      - name: Install debian dependenecies ${{ inputs._IMAGE_TYPE }}
```
Comment (JeGoi) [inline .github/workflows/reusable_sign_packages.yml]:
```suggestion
      - name: Install Debian dependencies ${{ inputs._IMAGE_TYPE }}
```

File: .github/workflows/reusable_sign_packages.yml [hunk 27]
```diff
@@ -0,0 +1,104 @@
...
+    steps:
+      - name: Dwnload artifactory ${{ inputs._IMAGE_TYPE }}
+        uses: actions/download-artifact@v3 
+        with:
+          name: ${{ env.ARTIFACTORY_NAME }}
+          path: /mnt
+        env:
+          ARTIFACTORY_NAME: ${{ inputs._IMAGE_TYPE == 'rhel8' && 'package-rpm' || 'package-deb' }}
+
+      - name: Install dependenecies ${{ inputs._IMAGE_TYPE }}
```
Comment (JeGoi) [inline .github/workflows/reusable_sign_packages.yml]:
```suggestion
      - name: Install dependencies ${{ inputs._IMAGE_TYPE }}
```

File: .github/workflows/reusable_sign_packages.yml [hunk 28]
```diff
@@ -0,0 +1,104 @@
...
+        run: dnf -y install rpm-sign python39 && python3.9 -m pip install -q  -U pip && pip install -q pynacl requests
+
+      - name: import private key ${{ inputs._IMAGE_TYPE }}
```
Comment (JeGoi) [inline .github/workflows/reusable_sign_packages.yml]:
```suggestion
      - name: Import private key ${{ inputs._IMAGE_TYPE }}
```

File: .github/workflows/reusable_sign_packages.yml [hunk 29]
```diff
@@ -0,0 +1,104 @@
...
+
+      - name: Check if the key was imported ${{ inputs._IMAGE_TYPE }}
+        run: |
+          rpm -q gpg-pubkey --qf '%{name}-%{version}-%{release} --> %{summary}\n'
+          gpg --list-keys
+
+      - name: Sign the package ${{ inputs._IMAGE_TYPE }}
+        run: rpm --define "_gpg_name support@inverse.ca" --define '_signature gpg' --addsign  /mnt/${{ inputs._PACKAGE_NAME }}*.rpm
+
+      - name: Verifiy the signarure of package ${{ inputs._IMAGE_TYPE }}
```
Comment (JeGoi) [inline .github/workflows/reusable_sign_packages.yml]:
```suggestion
      - name: Verify ${{ inputs._IMAGE_TYPE }} Package Signature
```

File: .github/workflows/reusable_unit_test.yml [hunk 30]
```diff
@@ -0,0 +1,34 @@
...
+        required: True
+        type: string
+
+jobs:
+  unit-test:
+    runs-on: packetfence-perl-package-build 
+    container:
+      image:  ${{ inputs._IMAGE_TYPE == 'rhel8' && 'redhat/ubi8:8.8' || 'debian:11.0'}}
+    steps:     
+      - name: download artifactory ${{ inputs._IMAGE_TYPE }}
```
Comment (JeGoi) [inline .github/workflows/reusable_unit_test.yml]:
```suggestion
      - name: Download artifactory ${{ inputs._IMAGE_TYPE }}
```

File: .github/workflows/packetfence-perl_build_image_package.yml [hunk 31]
```diff
@@ -0,0 +1,107 @@
...
+    runs-on: packetfence-perl-package-build
+    container:
+      image: ghcr.io/inverse-inc/packetfence/packetfence-perl-${{inputs._IMAGE_TYPE}}:${{ needs.build_image.outputs.package_version }}
+      credentials:
+        username: ${{ vars.USER_GITHUB }}
+        password: ${{ secrets.TOKEN_GITHUB }}
+      volumes:
+        - ${{ github.workspace }}/addons/packetfence-perl/:${{ inputs._WORKDIR  }}
+    steps:     
+      - name: "Build package ${{inputs._IMAGE_TYPE}}"
```
Comment (JeGoi) [inline .github/workflows/packetfence-perl_build_image_package.yml]:
```suggestion
      - name: Build package ${{inputs._IMAGE_TYPE}}
```

File: addons/packetfence-perl/psono.py [hunk 32]
```diff
@@ -0,0 +1,93 @@
...
+    data = json.dumps({
+        'api_key_id': api_key_id,
+        'secret_id': secret_id
+    })
+
+    decrypted_data = api_read_secret(data)
+
+    value = return_data(return_value, decrypted_data)
+
+    print(value)
```
Comment (JeGoi) [inline addons/packetfence-perl/psono.py:89]: Not sure I will print that.
Or only with a debug enabled or a flag.

File: containers/packetfence-perl/rhel8/Dockerfile_rhel8 [hunk 33]
```diff
@@ -0,0 +1,74 @@
+FROM redhat/ubi8
```
Comment (JeGoi) [inline containers/packetfence-perl/rhel8/Dockerfile_rhel8]: I think we should use a specific tag, not get the latest by default.

File: containers/packetfence-perl/debian/Dockerfile_debian [hunk 34]
```diff
@@ -0,0 +1,59 @@
+FROM debian:11
```
Comment (JeGoi) [inline containers/packetfence-perl/debian/Dockerfile_debian]: I think we should use a specific tag and NOT get the latest by default.

**Review by JeGoi — CHANGES_REQUESTED** [review]
I think docker tag version should be fix. Not use the latest by default.
Some names and variables to update.
Except that it looks pretty good!

File: addons/packetfence-perl/psono.py (hunk 32 above)
Comment (JeGoi) [reply addons/packetfence-perl/psono.py:89]: Ok I got it! I am fine with it

**Pull request merged**